File: android/src/main/java/com/aspire/secureencryption/SecureEncryptionPlugin.java
package com.aspire.secureencryption;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.PluginCall;
import com.getcapacitor.JSObject;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SecureEncryption")
public class SecureEncryptionPlugin extends Plugin {

    @PluginMethod
    public void encrypt(PluginCall call) {
        try {
            String plaintext = call.getString("plaintext");
            if (plaintext == null) {
                call.reject("Plaintext is required");
                return;
            }

            String ciphertext = SecureEncryption.encrypt(plaintext);
            JSObject result = new JSObject();
            result.put("ciphertext", ciphertext);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Encryption failed", e);
        }
    }

    @PluginMethod
    public void decrypt(PluginCall call) {  
        try {
            String ciphertext = call.getString("ciphertext");
            if (ciphertext == null) {
                call.reject("Ciphertext is required");
                return;
            }

            String plaintext = SecureEncryption.decrypt(ciphertext);
            JSObject result = new JSObject();
            result.put("plaintext", plaintext);
            call.resolve(result);
        } catch (Exception e) {
            call.reject("Decryption failed", e);
        }
    }
}

File: ios/Sources/SecureEncryptionPlugin/SecureEncryptionPlugin.swift
import Foundation
import Capacitor

@objc(SecureEncryptionPlugin)
public class SecureEncryptionPlugin: CAPPlugin {
    
    @objc func encrypt(_ call: CAPPluginCall) {
        guard let plaintext = call.getString("plaintext") else {
            call.reject("Plaintext is required")
            return
        }

        if let ciphertext = SecureEncryption.encrypt(plaintext) {
            call.resolve(["ciphertext": ciphertext])
        } else {
            call.reject("Encryption failed")
        }
    }

    @objc func decrypt(_ call: CAPPluginCall) {
        guard let ciphertext = call.getString("ciphertext") else {
            call.reject("Ciphertext is required")
            return
        }

        if let plaintext = SecureEncryption.decrypt(ciphertext) {
            call.resolve(["plaintext": plaintext])
        } else {
            call.reject("Decryption failed")
        }
    }
}
